sn_node/src/node/messaging/membership.rs

```diff
@@ -222,7 +223,17 @@ impl MyNode {
             cmds.extend(self.relocate_peers(churn_id, excluded_from_relocation)?);
         }
 
-        cmds.extend(self.trigger_dkg()?);
+        // In case the newly joined node is a relocated node, and we have enough elders,
+        // we don't try to promote it even it is qualified.
+        let full_of_elders = self.network_knowledge.elders().len() == elder_count();
+        let all_join_as_relocated = joining_nodes
```

Comment:
```suggestion
        let everything_joining_was_relocated = joining_nodes
```

`joined` ? the check is that everything that's joined us has relocated or not, right?

So, maybe for clarity: `we_should_find_new_elders = !(we_have_enough _elders && everything_joining_was_relocated)` ?